meteor/iterative.py

```diff
@@ -0,0 +1,270 @@
+from typing import Callable
+
+import numpy as np
+import pandas as pd
+import reciprocalspaceship as rs
+
+from .tv import TvDenoiseResult, tv_denoise_difference_map
+from .utils import (
+    average_phase_diff_in_degrees,
+    canonicalize_amplitudes,
+    complex_array_to_rs_dataseries,
+    rs_dataseies_to_complex_array,
+)
+
+
+def _project_derivative_on_experimental_set(
+    *,
+    native: np.ndarray,
+    derivative_amplitudes: np.ndarray,
+    difference: np.ndarray,
+) -> np.ndarray:
+    """
+    Project `derivative` onto the set of experimentally observed amplitudes,
+
+        Fh' = (D_F' + F) * [|Fh| / |D_F' + F|]
+
+    In English, the output is a complex-valued array that changes the derivative phase to ensure:
+
```

Comment:
Confusing docstring. Is derivative phase the "true" new state? In difference map code we have called derivative the second measured state